Alcatraz/Controllers/ATZPluginWindowController.m

```diff
@@ -277,4 +276,52 @@ - (void)addVersionToWindow {
     self.versionTextField.stringValue = @(ATZ_VERSION);
 }
 
+#pragma mark - Packages filtering
+
+- (void)updateInstallationStateSegmentedControl {
+    NSUInteger blacklistedPackagesCount = [self.packages filteredArrayUsingPredicate:[self blacklistedPredicate]].count;
+
+    if (blacklistedPackagesCount > 0) {
+        [self.installationStateSegmentedControl setSegmentCount:3];
+        NSString *label = [NSString stringWithFormat:@"Blocked (%lu)", blacklistedPackagesCount];
+        [self.installationStateSegmentedControl setLabel:label forSegment:2];
+    }
+    else {
+        [self.installationStateSegmentedControl setSegmentCount:2];
+        [self.installationStateSegmentedControl setSelectedSegment:0];
+        [self updatePredicate];
+    }
+}
+
+- (void)updatePredicate {
+    NSString *searchText = self.searchField.stringValue;
+    NSMutableArray* predicates = [[NSMutableArray alloc] initWithCapacity:3];
+    Class selectedPackageClass = [self segmentClassMapping][@([self.packageTypeSegmentedControl selectedSegment])];
+    if (selectedPackageClass)
+        [predicates addObject:[NSPredicate predicateWithFormat:CLASS_PREDICATE_FORMAT, selectedPackageClass]];
+
+    if (searchText.length > 0)
+        [predicates addObject:[NSPredicate predicateWithFormat:SEARCH_PREDICATE_FORMAT, searchText, searchText]];
+
+    if ([self.installationStateSegmentedControl selectedSegment] == 1)
+        [predicates addObject:[self installedPredicate]];
+    else if ([self.installationStateSegmentedControl selectedSegment] == 2)
+        [predicates addObject:[self blacklistedPredicate]];
```

Comment:
Remember when [apple didn't use braces](https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/)? I'd add them just in case :)